1.50.0/windowsservercore-1809/msvc/Dockerfile

```diff
@@ -0,0 +1,72 @@
+# escape=`
+
+# IMAGES BUILT FROM THIS DOCKERFILE ARE NOT TO BE SHARED ON PUBLIC DOCKER HUBs.
```

Comment:
The intent of this PR is to produce images that are to be shared on public Docker hubs, right?

Reply:
I'm not entirely sure at this point. I had completely missed this license issue until I was reading through @nelsonjchen's implementation. This issue was brought up by them before a few years ago and I didn't see any discussion about it.

I see value in providing a dockerfile for a base Rust on Windows image, but I don't know whether this repo is the right place for that. If we do keep this base dockerfile here, we would just need to make sure not to add it to the manifest file. (Which reminds me, I need to modify x.py to add the GNU builds to the manifest file...)